src/check_datapackage/check.py

```diff
@@ -304,12 +306,72 @@ def _handle_S_resources_x_schema_fields_x(
     return edits
 
 
+def _handle_S_resources_x_schema_fields_x_constraints_enum(
+    parent_error: SchemaError,
+    schema_errors: list[SchemaError],
+) -> SchemaErrorEdits:
+    """Only flag errors for the relevant field type and simplify errors."""
+
+    def _error_is_for_field_type() -> bool:
+        if not parent_error.parent:
+            return False
+        field_type: str = parent_error.parent.instance.get("type", "string")
+        if field_type not in FIELD_TYPES:
+            return False
+        schema_index = FIELD_TYPES.index(field_type)
+        return f"fields/items/oneOf/{schema_index}/" in parent_error.schema_path
+
+    edits = SchemaErrorEdits(remove=[parent_error])
+    errors_in_group = _get_errors_in_group(schema_errors, parent_error)
+
+    # Remove errors for other field types
+    if not _error_is_for_field_type():
+        edits.remove.extend(errors_in_group)
+        return edits
```

Comment:
Hmm, there are a too many returns in this function, which suggests maybe it's doing too much. Could you redesign in a way to not have so many early exits?